### PR TITLE
Adds a gitlab script for jiva-replica-failure

### PR DESCRIPTION
**Pull request by shashank855**
-Adds a gitlab script for jiva-replica-failure

### CODE REVIEW

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+cat deploy_run_test_jiva.yml
+
+bash ../Openshift-EE/utils/litmus_job_runner label='app:busybox-replica-failure-jiva' job=deploy_run_test_jiva.yml
+cd ..
+bash Openshift-EE/utils/dump_cluster_state;
+bash Openshift-EE/utils/event_updater jobname:app-replica-failure-jiva $test_name jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id"
+
+if [ "$?" != "0" ]; then
+python3 Openshift-EE/utils/result/result_update.py $job_id VJBK 4-chaos "Jiva Replica Failure" Fail $pipeline_id "$current_time" $commit_id $gittoken
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 can you give the test description even more meaningful.

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+present_dir=$(pwd)
+echo $present_dir
+
+bash Openshift-EE/utils/e2e-cr jobname:app-replica-failure-jiva jobphase:Waiting init_time:"$current_time" jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id"
+bash Openshift-EE/utils/e2e-cr jobname:app-replica-failure-jiva jobphase:Running
+
+echo "*******Deploying Busybox-sts Application****"
+
+test_name=$(bash Openshift-EE/utils/generate_test_name testcase=busybox-provision-app-replica-failure metadata="")
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 check the testcase name. Is it correct?

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+if [ "$?" != "0" ]; then
+python3 Openshift-EE/utils/result/result_update.py $job_id VJBK 4-chaos "Jiva Replica Failure" Fail $pipeline_id "$current_time" $commit_id $gittoken
+exit 1;
+fi
+
+################
+# LitmusBook 2 #
+################
+
+run_id="jiva";test_name=$(bash Openshift-EE/utils/generate_test_name testcase=volume-replica-failure metadata=${run_id})
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 this test case name must be the same as there in the test_vars.
`https://github.com/openebs/litmus/blob/master/apps/percona/chaos/openebs_volume_replica_failure/test_vars.yml`

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+            value: '"$run_id"'\
+' run_rep_fail_test_jiva.yml
+
+echo "Running the litmus test for Busybox Deployment Scaleup.."
+cat run_rep_fail_test_jiva.yml
+
+bash ../Openshift-EE/utils/litmus_job_runner label='name:openebs-volume-replica-failure' job=run_rep_fail_test_jiva.yml
+cd ..
+bash Openshift-EE/utils/dump_cluster_state;
+bash Openshift-EE/utils/event_updater jobname:app-replica-scaleup-jiva $test_name jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id"
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 can you use a  job name like volume-rep-failure

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+if [ "$?" != "0" ]; then
+python3 Openshift-EE/utils/result/result_update.py $job_id VJBK 4-chaos "Jiva Replica Failure" Fail $pipeline_id "$current_time" $commit_id $gittoken
+exit 1;
+fi
+
+echo "********Deprovisioning Busybox Application*******"
+
+test_name=$(bash Openshift-EE/utils/generate_test_name testcase=busybox-deprovision-app-scale-jiva metadata="")
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 check the testcase name here

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+echo $test_name
+
+################
+# LitmusBook 3 #
+################
+
+cd litmus
+cp apps/busybox/deployers/run_litmus_test.yml deprovision_scaleup_jiva.yml
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 can you use a different file name to copy the run_litmus_test spec.

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+  ----------------------------------------------------------------------------------------------------------------------
+EOF
+
+sed -i -e 's/imagePullPolicy: Always/imagePullPolicy: IfNotPresent/g' \
+-e 's/generateName: litmus-busybox-deploy/generateName: busybox-provision-app-rep-failure/g' \
+-e 's/value: app-busybox-ns/value: app-replica-fail-jiva/g' \
+-e 's/value: openebs-busybox/value: openebs-busybox-rep-kill/g' \
+-e 's/app: busybox-litmus/app: busybox-replica-scale-jiva/g' \
+-e 's/value: '\''app=busybox-sts'\''/value: '\''app=busybox-rep-fail-jiva'\''/g' \
+-e 's/value: openebs-cstor-sparse/value: openebs-jiva-default/g' deploy_run_test_jiva.yml
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 can you deploy busybox as a deployment.

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+commit_id=$(echo $3)
+case_id=VJBK
+
+time="date"
+current_time=$(eval $time)
+
+present_dir=$(pwd)
+echo $present_dir
+
+bash Openshift-EE/utils/e2e-cr jobname:app-replica-failure-jiva jobphase:Waiting init_time:"$current_time" jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id"
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 can you use a  job name like volume-rep-failure

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+
+rc_val=$(echo $?)
+
+# testResult=$(kubectl get litmusresult ${test_name} --no-headers -o custom-columns=:spec.testStatus.result)
+# if [ "$rc_val" != "0" ]; then
+# exit 1;
+# fi
+
+current_time=$(eval $time)
+bash Openshift-EE/utils/e2e-cr jobname:app-replica-scaleup-jiva jobphase:Completed end_time:"$current_time" jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id" test_result:Pass
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 can you use a  job name like volume-rep-failure

Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 can you use a  job name like volume-rep-failure

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+-e 's/value: provision/value: deprovision/g' deprovision_scaleup_jiva.yml
+
+
+echo "Running the litmus test for Busybox Deployment Scaleup.."
+cat deprovision_scaleup_jiva.yml
+
+bash ../Openshift-EE/utils/litmus_job_runner label='app:busybox-deprovision-rep-fail' job=deprovision_scaleup_jiva.yml
+cd ..
+bash Openshift-EE/utils/dump_cluster_state;
+bash Openshift-EE/utils/event_updater jobname:app-replica-scaleup-jiva $test_name jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id"
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 can you use a  job name like volume-rep-failure

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+ | Litmus job label  | label  | app: busybox-litmus                 | app: busybox-replica-failure-jiva                |
+  ----------------------------------------------------------------------------------------------------------------------
+EOF
+
+sed -i -e 's/imagePullPolicy: Always/imagePullPolicy: IfNotPresent/g' \
+-e 's/generateName: litmus-busybox-deploy/generateName: busybox-provision-app-rep-fail-jiva/g' \
+-e 's/value: app-busybox-ns/value: app-replica-fail-jiva/g' \
+-e 's/value: statefulset/value: deployment/g' \
+-e 's/value: openebs-busybox/value: openebs-busybox-rep-kill/g' \
+-e 's/app: busybox-litmus/app: busybox-replica-scale-jiva/g' \
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 Can you change the litmus job label.

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+  ---------------------------------------------------------------------------------------------------------------------|
+ | appLabel          | env    | name=percona                        | app=busybox-rep-fail-jiva                        | 
+ | appPVC            | env    | value: percona-mysql-claim          | value: openebs-busybox-rep-kill                  |
+ | appNamespace      | env    | app-percona-ns                      | app-replica-fail-jiva                            | 
+  ----------------------------------------------------------------------------------------------------------------------
+EOF
+
+sed -i -e 's/name=percona /app=busybox-rep-fail-jiva/g' \
+-e 's/value: app-percona-ns/value: app-replica-fail-jiva/g' \
+-e 's/value: percona-mysql-claim/value: app-replica-fail-jiva/g' run_rep_fail_test_jiva.yml
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 can you use the application PVC name correctly.

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/KLPX-Jiva-replica-network-delay/jiva-replica-netork-delay [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+ | Litmus job label  | label  | app: busybox-litmus                 | app: busybox-deprovision-rep-network                     |
+  -----------------------------------------------------------------------------------------------------------------------------
+EOF
+
+sed -i -e 's/generateName: litmus-busybox-deploy/generateName: busybox-deprovision-rep-network-delay-jiva/g' \
+-e 's/value: app-busybox-ns/value: rep-network-delay-jiva/g' \
+-e 's/app: busybox-litmus/app: busybox-deprovision-rep-fail/g' \
+-e 's/value: '\''app=busybox-sts'\''/value: '\''app=busybox-rep-network-jiva'\''/g' \
+-e 's/value: openebs-cstor-sparse/value: openebs-jiva-default/g' \
+-e 's/value: provision/value: deprovision/g' deprovision_rep_network_jiva.yml
```
[the remark this answers is not in the document]
Comment (shashank855) [reply Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/KLPX-Jiva-replica-network-delay/jiva-replica-netork-delay]: @nsathyaseelan I have made the changes.

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/KLPX-Jiva-replica-network-delay/jiva-replica-netork-delay [hunk 14]
```diff
@@ -0,0 +1,175 @@
...
+time="date"
+current_time=$(eval $time)
+
+present_dir=$(pwd)
+echo $present_dir
+
+bash Openshift-EE/utils/e2e-cr jobname:app-replica-network-delay-jiva jobphase:Waiting init_time:"$current_time" jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id"
+bash Openshift-EE/utils/e2e-cr jobname:app-replica-network-delay-jiva jobphase:Running
+
+echo "*******Deploying Busybox-sts Application****"
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/KLPX-Jiva-replica-network-delay/jiva-replica-netork-delay]: @shashank855 deploying stateful set of deployment

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 15]
```diff
@@ -0,0 +1,174 @@
...
+commit_id=$(echo $3)
+case_id=VJBK
+
+time="date"
+current_time=$(eval $time)
+
+present_dir=$(pwd)
+echo $present_dir
+
+bash Openshift-EE/utils/e2e-cr jobname:app-replica-failurse-jiva jobphase:Waiting init_time:"$current_time" jobid:"$job_id" pipelineid:"$pipeline_id" testcaseid:"$case_id"
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 The job name here is volume replica failure.

File: Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure [hunk 16]
```diff
@@ -0,0 +1,174 @@
...
+bash Openshift-EE/utils/e2e-cr jobname:app-replica-failure-jiva jobphase:Running
+
+echo "*******Deploying Busybox-sts Application****"
+
+test_name=$(bash Openshift-EE/utils/generate_test_name testcase=busybox-provision-app-rep-fail-jiva metadata="")
```
Comment (nsathyaseelan) [inline Openshift-EE/pipelines/OpenEBS-base/stages/4-chaos/VJBK-Jiva-replica-failure/jiva-replica-failure]: @shashank855 the test case should be busybox-provision-app-replica-fail-jiva

**Pull request merged**